### DISCUSSION TITLE
How to handle return value of SnowflakeOperator in Airflow

### DISCUSSION THREAD

**Discussion by christianklat** [apache/airflow#28258] [Q&A]
Hello!

I have already asked Stackoverflow regarding my problem, however I have not received an answer for over 2 weeks now. 
Therefore, I will just copy my post here and hope that there might be someone in this discussion board who can help me out. 

Would really appreciate it!!

> I'm currently experimenting with Airflow for monitoring tasks regarding Snowflake and I'd like to execute a simple DAG with one task that pushes a SQL query to in Snowflake and should check the returned value that should be a number to be greater than a defined threshold.
> 
> So the following is basically my sql Statement in the DAG definition:
> 
> ```
> query_check = """select COUNT(*)
> FROM (select CASE WHEN NAME LIKE '%SW_PRODUCTFEED%' THEN 'PRODUCTFEED'
>         ELSE NULL END AS TASKTREE_NAME
>        , NAME 
>        , STATE
>        , ERROR_MESSAGE
>        , SCHEDULED_TIME
>        , QUERY_START_TIME
>        , NEXT_SCHEDULED_TIME
>   from table(TEST_DB.INFORMATION_SCHEMA.task_history())
>   where TASKTREE_NAME IS NOT NULL
>   qualify DENSE_RANK() OVER (PARTITION BY TASKTREE_NAME ORDER BY to_date(SCHEDULED_TIME) desc) < 3
>   order by scheduled_time desc);"""
> ```
> 
> Then the following is the definition of the DAG and the task within it:
> 
> ```
> with dag:
>     query1_exec = SnowflakeCheckOperator(
>         task_id="snowflake_check_task_history",
>         sql=query_check,
>         params={
>             "check_name": "number_rows",
>             "check_statement": "count >=1"
>         },
>         conn_id="Snowflake_test"
>     )
> 
> query1_exec
> ```
> 
> I'd like to use the SnowflakeCheckOperator to check the returned value from the query if it's greater than 1
> 
> However, it seems that Snowflake or the SnowflakeOperator in that case is returning the result of the query in a dict object, like so:
> 
> `Record: {'COUNT(*)': 10}
> `
> Therefore the check always results in a true statement because the SnowflakeCheckOperator isn't checking against the value of the `Record["Count"]` but something else.
> 
> Now my question is how to handle the return value so that the check is evaluated against right value? Is it possible to change the format of the return value? Or maybe get access to the value of the key of the dict object?

**Comment by potiuk** [comment]
This should be fixed in 4.0.2 version of Snowflake Provider. https://airflow.apache.org/docs/apache-airflow-providers-snowflake/stable/index.html#id1 - 4.0.1 and 4.0.0 version were yanked precisely because problems with this. 

Please check 4.0.2 version and see if it works as expected.
